k8s/tools/cleanup-pubsub-gke.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+# Turbinia GKE cleanup script
+# This script can be used to cleanup the Turbinia stack within GKE PubSub. Note that
+# this script will not disable any APIs to avoid outage with any other applications 
+# deployed within the project. 
+# Requirements:
+# - have 'gcloud'installed.
+# - autheticate against your GCP project with "gcloud auth login"
+# - optionally have the GCP project set with "gcloud config set project [you-project-name]"
+#
+# Use --help to show you commands supported.
+
+set -o posix
+set -e
+
+# Source cluster config to pull specs to create cluster from. Please review
+# the config file and ensure the parameters are set to the cluster you are 
+# intending to cleanup
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+source $DIR/.clusterconfig
+cd $DIR/..
+
+if [[ "$*" == *--help ]] ; then
+  echo "Turbinia cleanup script for Turbinia within Kubernetes"
+  echo "Options:"
+  echo "--no-gcloud-auth               Do not use gcloud authentication and service key instead"
+  echo "--no-cloudfunctions            Do not cleanup Turbinia Cloud Functions"
+  echo "--no-datastore                 Do not cleanup Turbinia Datastore"
+  echo "--no-filestore                 Do not cleanup Turbinia Filestore share"
+  exit 1
+fi
+
+# Check configured gcloud project
+if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+  DEVSHELL_PROJECT_ID=$(gcloud config get-value project)
+  ERRMSG="ERROR: Could not get configured project. Please either restart "
+  ERRMSG+="Google Cloudshell, or set configured project with "
+  ERRMSG+="'gcloud config set project PROJECT' when running outside of Cloudshell."
+  if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+  echo "Environment variable \$DEVSHELL_PROJECT_ID was not set at start time "
+  echo "so attempting to get project config from gcloud config."
+  echo -n "Do you want to use $DEVSHELL_PROJECT_ID as the target project? (y / n) > "
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# Delete the cluster
+echo "Deleting cluster $CLUSTER_NAME"
+gcloud -q --project $DEVSHELL_PROJECT_ID container clusters delete $CLUSTER_NAME --zone $ZONE
+
+# Delete the GCS storage bucket
+echo "Deleting GCS storage bucket gs://$INSTANCE_ID"
```

Comment:
This script is going to do a lot of destructive / irrecoverable actions.  Maybe we can add another check at the start of the script to warn the user and make sure they know that this will delete all output files and all logs, etc, and force them to type "delete all" or something like that?

Also, it's probably worth adding something like a `--no-gcs` flag not to delete the GCS bucket.

Reply:
Added a note that will prompt user to confirm